src/api/index.md

```diff
@@ -0,0 +1,277 @@
+			} ),
+			controlType( {
+				type: 'bold',
+				title: 'Make Bold',
+				displayArea: 'inline'
+			} ),
+			controlType( {
+				type: 'italics',
+				title: 'Italicize',
+				displayArea: 'inline'
+			} ),
+			controlType( {
+				type: 'strikethrough',
+				title: 'Strikethrough',
+				displayArea: 'inline'
+			} ),
+			controlType( {
+				type: 'underline',
+				title: 'Underline',
+				displayArea: 'inline'
+			} )
+		]
+	]
+	create: ( blockType = this ) => textBlock( blockType )
+}
+
+const textBlockType = blockType( textBlockTypeProperties )
+
+// Create a text block.
+const aTextBlock = textBlockType.create()
+
+// Then grab the DOM node it creates.
+const myTextDOMNode = aTextBlock.render( 'Blocks are neat.' )
+```
+
+### Creating a textBlock
+
+```js
+// Example text block.
+const textBlock = blockType => {
+    return block ( {
+        /**
+         * State and props would be objects that would be passed in via data held somewhere. i.e. the UI
+         *
+         * The render method is defined by you, so you can make it work the way you want it to.
+         */
+        render: ( state = {}, props = {} ) => {
+            // Do some stuff relating to the blockType for this block.
+
+            // Render something. Could be DOM, vDOM, React, TinyMCE, JSON, XML, HTML, whatever you want.
+            // Here is DOM:
+            if ( typeof document !== 'undefined' ) {
+              pElement = document.createElement( 'p' )
+              textNode = document.createTextNode( props.text )
+              pElement.appendChild( textNode )
+              return pElement
+            }
+
+            // Here is vDOM:
+            if ( typeof h !== 'undefined' ) {
+                return h( 'p', '.wp-text', [
+                    props.text
+                ] )
+            }
+
+            // React.js could work something similar to this.
+            if ( typeof SomeReactComponent !== 'undefined' ) {
+                return <SomeReactComponent state={state} props={props} />
+            }
+        }
+    } )
+}
+```
+
+Because the blockType is encapsulated within the reference to this block, you
+could do interesting things based on properties of the blockType.
+
+### Cool ideas & the importance of separation of the view layer.
+
+```js
+// blockProperties would include the render() method.
+const textBlock = block( blockProperties )
+
+// textTypeProperties could contain a create method returning textBlock().
+const textType = blockType( textTypeProperties )
+
+const aTextBlock = textType.create()
+const anotherTextBlock = textType.create( textType( withDifferentPropertiesIfYouWant ) )
+
+// containerTypeProperties's create() method would create the actual block.
+const containerType = blockType( containerTypeProperties )
+
+// containerProperties's render() method could provide a way to render both aTextBlock and anotherTextBlock inside of it.
+const blockContainer = block( containerProperties )
+
+let container = containerType(
```

Comment:
Yup :).  I imagine at some point nested blocks will be something people want, i.e. multi column layouts, galleries.  A container would serve as a block that just renders blocks contained inside of it.  It's controls would be to add more blocks or remove blocks.  Since there is an aversion to doing nested blocks upfront, it is good to at least map out how that would work with an API.  I wouldn't worry about this too much, just an idea.